src/store/channels/saga.ts

```diff
@@ -136,7 +145,63 @@ export function* roomUnfavorited(action) {
   }
 }
 
+export function* publishUserTypingEvent(roomId) {
+  try {
+    const channel = yield select(rawChannelSelector(roomId));
+    if (!channel) {
+      return;
+    }
+
+    const currentUserLastTypingEventPublishedAt = channel.currentUserLastTypingEventPublishedAt;
+    const now = Date.now();
+    if (currentUserLastTypingEventPublishedAt && now - currentUserLastTypingEventPublishedAt <= USER_TYPING_TIMEOUT) {
+      // not enough time has passed since last typing event
+      return;
+    }
```

Comment:
Is this really what we want? Assuming the matrix typing event timeout that we passed in means that it'll clear the typing status after that timeout.... if I were to type non-stop for 4999ms seconds and then pause for a second, the typing status would clear. Should it clear if I _just_ typed something a couple ms ago?

Maybe on line 203 we should just use this instead: https://redux-saga.js.org/docs/api#debouncems-pattern-saga-args

```
yield debounce(1000, SagaActionTypes.UserTypingInRoom, publishUserTypingEvent)
```

Doesn't even really matter if the room changes because the time between typing in one room, switching rooms, and typing in the next room will pretty much cross the 1 second debounce after a few characters anyway.

That removes the complexity of managing this extra state of the "last published" time.

Reply:
as discussed on call, updated to use `leadingDebounce`